apps/src/templates/projects/ReportAbusePopUp.jsx

```diff
@@ -0,0 +1,279 @@
+      'authenticity_token',
+      RailsAuthenticityToken.getRailsCSRFMetaTags().token
+    );
+    formData.append('channel_id', this.props.projectData.channel);
+    formData.append('abuse_url', this.props.abuseUrl);
+    formData.append('g-recaptcha-response', this.token);
+
+    const checkedCheckboxNames = this.state.checkboxes
+      .filter(checkbox => checkbox.checked)
+      .map(checkbox => checkbox.key.replace(/-/g, ' '));
+    formData.append('abuse_type', checkedCheckboxNames);
+
+    fetch('/report_abuse_pop_up', {
+      method: 'POST',
+      body: formData,
+    })
+      .then(response => {
+        if (response.ok) {
+          this.setState({
+            showReportConfirmation: true,
+          });
+          this.props.onReport();
+        } else if (response.status === 403) {
+          this.onCaptchaExpiration();
+        }
+      })
+      .catch(error => {
+        this.setState({
+          submitButtonEnabled: false,
+        });
+        console.error('Error sending report data:', error);
+      });
+  }
+
+  onCaptchaExpiration() {
+    this.setState({
+      submitButtonEnabled: false,
+      captchaCompleted: false,
+    });
+  }
+
+  handleCheckboxChange = checkboxKey => {
+    this.setState(prevState => {
+      const updatedCheckboxes = prevState.checkboxes.map(checkbox =>
+        checkbox.key === checkboxKey
+          ? {...checkbox, checked: !checkbox.checked}
+          : checkbox
+      );
+
+      const showRecaptcha = updatedCheckboxes.some(
+        checkbox => checkbox.checked
+      );
+
+      return {
+        checkboxes: updatedCheckboxes,
+        showRecaptcha,
+      };
+    });
+  };
+
+  createCaptchaScript() {
+    const script = document.createElement('script');
+    script.src = 'https://www.google.com/recaptcha/api.js';
+    script.id = 'captcha';
+    script.async = true;
+    script.defer = true;
+    window.onCaptchaSubmit = token => this.onCaptchaVerification(token);
+    window.onCaptchaExpired = () => this.onCaptchaExpiration();
+    script.onload = () => this.setState({loadedCaptcha: true});
+    document.body.appendChild(script);
+  }
+
+  render() {
+    const {
+      showRecaptcha,
+      checkboxes,
+      submitButtonEnabled,
+      showReportConfirmation,
+    } = this.state;
+
+    const captchaSiteKey = this.props.recaptchaSiteKey;
+
+    return (
+      <AccessibleDialog className={style.popUp} onClose={this.cancel}>
+        {showReportConfirmation ? (
+          <div className={style.submitConfirmation}>
+            <h3>{i18n.thankyou()}!</h3>
+            <p>{i18n.thankYouForReport()}</p>
+            <Button
+              onClick={this.cancel}
+              text={'Continue'}
+              color={Button.ButtonColor.brandSecondaryDefault}
+            />
+          </div>
+        ) : (
+          <div>
+            <div className={style.title}>
+              <h3 style={{margin: 0}}>{i18n.reportAbuse()}</h3>
+              <button
+                type="reset"
```

Comment:
Not super familiar with this, but I don't think this quite the right use of the button type `reset`, docs [here](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/button)...FWIW I see 0 other usages of type="reset" in our code base